### PR TITLE
feat: prepare for react 18 and redux 5

### PR DESCRIPTION
**Pull request by JAdshead**
## Description

This prepares Holocron to be used with React 18 and Redux 5.

- Test suite migrated to RTL, removing enzyme.
- Redux and react redux added to peer deps

## Motivation and Context

Allow holocron to be used with React 18 and Redux 5

## How Has This Been Tested?

Unit tests ran with current React dev versions `16.11.0`

I also installed React 18, Rexux 5 and React Redux 9 and ran the tests.
The only change required to get passing was to update `testing-library-react` to `^14`. 

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] My changes are in sync with the code style of this project.
- [ ] There aren't any other open Pull Requests for the same issue/update.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] This change impacts caching for client browsers.
- [ ] This change adds additional environment variable requirements for Holocron users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using Holocron?

This allows consumers to use newer versions of Redux and React Redux.

### CODE REVIEW

File: packages/holocron-module-route/__tests__/ModuleRoute.spec.jsx [hunk 1]
```diff
@@ -13,24 +13,24 @@
...
 
 import React from 'react';
-// eslint erroneously believes enzyme is not a dev dependency
-// eslint-disable-next-line import/no-extraneous-dependencies -- only used in tests
-import { shallow } from 'enzyme';
 
 import ModuleRoute from '../src/ModuleRoute';
+// eslint erroneously believes '@testing-library/react' is not a dev dependency
+// eslint-disable-next-line import/no-extraneous-dependencies -- only used in tests
+const { render } = require('@testing-library/react');
```
Comment (Matthew-Mallimo) [inline packages/holocron-module-route/__tests__/ModuleRoute.spec.jsx:20]: is this lint error only in your editor or does it show up when running eslint in a terminal? Sometimes I find the editor gets confused but actual linter doesn't

File: packages/holocron/__tests__/holocronModule.spec.jsx [hunk 2]
```diff
@@ -426,17 +422,13 @@ describe('holocronModule', () => {
...
     await sleep(1);
-    wrapper.update();
-    expect(wrapper.find(TestComponent).prop('moduleLoadStatus')).toEqual(
-      'error'
-    );
-    wrapper.unmount();
+    expect(screen.getByRole('status')).toHaveTextContent('error');
   });
 
   it('should not try to setStatus if it is not mounted', () => {
```
Comment (Matthew-Mallimo) [inline packages/holocron/__tests__/holocronModule.spec.jsx:434]: Can we remove this test case as it has no assertions?

File: packages/holocron/package.json [hunk 3]
```diff
@@ -30,21 +30,21 @@
...
+    "redux-thunk": "^3.1.0",
     "require-from-string": "^1.2.1",
     "reselect": "^4.0.0",
     "semver": "^7.3.8",
     "shallowequal": "^1.1.0",
     "ssri": "^7.0.1"
   },
   "devDependencies": {
+    "redux": "^4.0.4",
+    "react-redux": "^7.1.1",
```
Comment (Matthew-Mallimo) [inline packages/holocron/package.json:47]: our peerdeps requires `>=9.1.0` but here we are only using `7.1.1`

File: packages/holocron/package.json [hunk 4]
```diff
@@ -30,21 +30,21 @@
...
-    "redux-thunk": "^2.1.0",
+    "redux-thunk": "^3.1.0",
     "require-from-string": "^1.2.1",
     "reselect": "^4.0.0",
     "semver": "^7.3.8",
     "shallowequal": "^1.1.0",
     "ssri": "^7.0.1"
   },
   "devDependencies": {
+    "redux": "^4.0.4",
```
Comment (Matthew-Mallimo) [inline packages/holocron/package.json:46]: our peerdeps requires `>=5.0.1` but here we are only using `4.0.4`

File: packages/holocron-module-route/__tests__/ModuleRoute.spec.jsx (hunk 1 above)
Comment (JAdshead) [reply packages/holocron-module-route/__tests__/ModuleRoute.spec.jsx:20]: this is an eslint error

File: packages/holocron/package.json (hunk 3 above)
Comment (JAdshead) [reply packages/holocron/package.json:47]: ah that a miss

File: packages/holocron/package.json (hunk 4 above)
Comment (JAdshead) [reply packages/holocron/package.json:46]: [89f4184](https://github.com/americanexpress/holocron/pull/172/commits/89f418488ef1dbea938c7d3a2710987da0a5bf15)

File: packages/holocron/package.json (hunk 3 above)
Comment (JAdshead) [reply packages/holocron/package.json:47]: [89f4184](https://github.com/americanexpress/holocron/pull/172/commits/89f418488ef1dbea938c7d3a2710987da0a5bf15)

File: jest.config.js [hunk 5]
```diff
@@ -26,4 +26,5 @@ module.exports = {
   transform: {
     '^.+\\.[t|j]sx?$': 'babel-jest',
   },
+  snapshotSerializers: [],
```
Comment (giulianok) [inline jest.config.js:29]: why is this needed?

Comment (JAdshead) [reply jest.config.js:29]: This was added during the RTL migration. https://github.com/americanexpress/holocron/pull/166

**Review by Matthew-Mallimo — APPROVED** [review]
